core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -195,10 +198,9 @@ public InternalEngine(EngineConfig engineConfig) {
                 historyUUID = loadOrGenerateHistoryUUID(writer, engineConfig.getForceNewHistoryUUID());
                 Objects.requireNonNull(historyUUID, "history uuid should not be null");
                 indexWriter = writer;
-                translog = openTranslog(engineConfig, writer, translogDeletionPolicy, () -> seqNoService.getGlobalCheckpoint());
-                assert translog.getGeneration() != null;
                 // we can only do this after we generated and committed a translog uuid. other wise the combined
                 // retention policy, which listens to commits, gets all confused.
+                refreshTranslogUUIDIfNeeded(writer, translog);
```

Comment:
move this line above the comment? The comment applies to persistHistoryUUIDIfNeeded